source/Calamari.Common/Plumbing/Variables/ScriptVariables.cs

```diff
@@ -28,10 +28,9 @@ public static string GetLibraryScriptModuleName(string variableName)
             return variableName.Replace("Octopus.Script.Module[", "").TrimEnd(']');
         }
         
-        public static string FormatScriptNameForPhysicalFilesystem(string scriptName)
+        public static string FormatScriptName(string scriptName)
         {
-            var fileSystem = CalamariPhysicalFileSystem.GetPhysicalFileSystem();
-            return fileSystem.RemoveInvalidFileNameChars(scriptName); 
+            return new string(scriptName.Where(x => char.IsLetterOrDigit(x) || x == '_').ToArray());
```

Comment:
This is identical to the pre-existing logic, with the addition of allowing underscores.